src/gt4py/next/ffront/past_passes/type_deduction.py

```diff
@@ -108,7 +108,7 @@ def visit_Subscript(self, node: past.Subscript, **kwargs: Any) -> past.Subscript
         value = self.visit(node.value, **kwargs)
         return past.Subscript(
             value=value,
-            slice_=self.visit(node.slice_, **kwargs),
+            slice_=node.slice_,  # TODO: I don't think we are using this type
```

Comment:
```suggestion
            slice_=node.slice_,
```
check and remove

Reply:
We don't have a type for `past.Slice`, but the visit here also types & validates its children (e.g `1+1.` in a slice bound would error here). For that reason I consider unconditionally visiting all children a good approach. The role of `ts.DeferredType` and `None` in the type system is however not well specified. I would postpone that discussion to the point where we also have time to think about generics / "templated" / partial types, because the discussion is related.

Reply:
What are you proposing to do? Accept `None` in TupleType elements, this proposed solution or add `visit_Slice` and return `DeferredType`.

Reply:
Ah I overlooked that `slice_` is a tuple of slices (maybe the attr name is not so great?). `visit_Slice` with `DeferredType` makes sense to me even though this would add another violation of all-type-attrs-of-types-are-concrete rule of a `ts.TypeSpec`. Otherwise introduction of a slice type makes sense to me too.

Reply:
done